pkg/logs/input/file/position.go

```diff
@@ -12,25 +12,82 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
 )
 
+// TailingMode type
+type TailingMode uint8
+
+// Tailing Modes
+const (
+	ForceBeginning = iota
+	ForceEnd
+	Beginning
+	End
+)
+
+// Tailing mode string representation
+const (
+	ForceBeginningStr = "forceBeginning"
+	ForceEndStr       = "forceEnd"
+	BeginningStr      = "beginning"
+	EndStr            = "end"
+)
+
+var tailingModeToStringRepresentations = map[TailingMode]string{
+	ForceBeginning: ForceBeginningStr,
+	ForceEnd:       ForceEndStr,
+	Beginning:      BeginningStr,
+	End:            EndStr,
+}
+
+// TailingModeFromString parses a string and returns a corresponding tailing mode, default to End if not found
+func TailingModeFromString(mode string) (TailingMode, bool) {
+	for m, mStr := range tailingModeToStringRepresentations {
+		if mStr == mode {
+			return m, true
+		}
+	}
+	return End, false
+}
+
+// TailingModeToString returns seelog string representation for a specified tailing mode. Returns "" for invalid tailing mode.
+func (mode TailingMode) String() string {
+	modeStr, ok := tailingModeToStringRepresentations[mode]
+	if ok {
+		return modeStr
+	}
+	return ""
+}
+
 // Position returns the position from where logs should be collected.
-func Position(registry auditor.Registry, identifier string, tailFromBeginning bool) (int64, int, error) {
+func Position(registry auditor.Registry, identifier string, m TailingMode) (int64, int, error) {
 	var offset int64
 	var whence int
 	var err error
+
 	value := registry.GetOffset(identifier)
+
 	switch {
+	case m == ForceBeginning:
```

Comment:
do we really want those force-thing ? We should make it clear that force can lead to loss or duplication which has impact on the user experience.

Reply:
I totally agree, the documentation that will come with this new attribute will only account for `end` and `beginning`  (as per https://docs-staging.datadoghq.com/prognant/agent_log_start_position/agent/logs/?tab=tailfiles). The `forceBeginning` value is used internally, and the `forceEnd` was added for concistency.

Reply:
We might want to add a log line in the agent when we're forcing "forceEnd" by switching from "beginning" to "end"